### PR TITLE
refactor: type consolidation

### PR DESCRIPTION
**Pull request by mshanemc**
uses types for sfProject, PackageDir etc from sfdx-core
refactor some un-bundleable geLogger stuff
othr basic TS cleanup (remove assertions, simplify types)

### CODE REVIEW

File: test/package/ancestry.nut.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
+/*
```
Comment (mshanemc) [inline test/package/ancestry.nut.ts:1]: this is completely independent from the e2e nut so I split it out (can run parallel)

File: test/package/packageTest.nut.ts [hunk 2]
```diff
@@ -128,6 +124,7 @@ describe('Integration tests for @salesforce/packaging library', () => {
       const projectFile = JSON.parse(dxPjsonData) as ProjectJson;
       expect(projectFile).to.have.property('packageDirectories').with.length(1);
       expect(projectFile.packageDirectories[0]).to.include.keys(['package', 'versionName', 'versionNumber']);
+      assert(isPackagingDirectory(projectFile.packageDirectories[0]));
       expect(projectFile.packageDirectories[0].package).to.equal(pkgName);
```
Comment (mshanemc) [inline test/package/packageTest.nut.ts:128]: these all presume packageDirectories[0] is the flavor that has the full packaging info.  asserting that via the typeguard makes everything downstream of that assertion type-aware

File: src/package/packageVersionCreate.ts [hunk 3]
```diff
@@ -649,16 +650,19 @@ export class PackageVersionCreate {
         if (!packageName) throw messages.createError('errorMissingPackage', [this.options.packageId]);
       }
       packageObject = this.project.findPackage(
-        (namedPackageDir) => namedPackageDir.package === packageName || namedPackageDir.name === packageName
+        (namedPackageDir) =>
+          (isPackagingDirectory(namedPackageDir) && namedPackageDir.package === packageName) ||
```
Comment (WillieRuemmele) [inline src/package/packageVersionCreate.ts:654]: I don't think this logic matches, should it be 
`isPackagingDirectory(namedPackageDir) &&(namedPackageDir.package === packageName || namedPackageDir.name === packageName)`?

File: src/package1/package1Version.ts [hunk 4]
```diff
@@ -172,3 +136,32 @@ export class Package1Version implements IPackageVersion1GP {
     return (await this.connection.tooling.query<PackagingSObjects.MetadataPackageVersion>(query)).records;
   }
 }
+
+const packageUploadPolling = async (
```
Comment (WillieRuemmele) [inline src/package1/package1Version.ts:140]: I believe this is only applicable to 1GP packages, and why it was kept under the `Package1Version` class.  I think keeping that clarity to the consumer is good

**Review by WillieRuemmele — CHANGES_REQUESTED** [review]

File: src/package1/package1Version.ts (hunk 4 above)
Comment (mshanemc) [reply src/package1/package1Version.ts:140]: I'm not following.  it's not exported, so there shouldn't be any consumers outside the package1Version module, right?

**Review by WillieRuemmele — APPROVED** [review]

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**